app/src/main/java/sdp/moneyrun/MenuActivity.java

```diff
@@ -222,6 +222,72 @@ public void onButtonSwitchToUserProfileActivity(View view) {
         setPutExtraArguments(playerProfileIntent);
         startActivity(playerProfileIntent);
     }
+
+
+    public void onButtonShowQuestionPopupWindowClick(View view, Boolean focusable, int layoutId, Riddle riddle) {
+
+        PopupWindow popupWindow = onButtonShowPopupWindowClick(view, focusable, layoutId);
+        TextView tv = popupWindow.getContentView().findViewById(R.id.question);
+        int correctId = 0;
+
+        //changes the text to the current question
+        tv.setText(riddle.getQuestion());
+
+        int[] buttonIds = {R.id.question_choice_1, R.id.question_choice_2, R.id.question_choice_3, R.id.question_choice_4};
+        TextView buttonView = tv;
+
+        //Loops to find the ID of the button solution and assigns the text to each button
+        /*for (int i = 0; i < 4; i++){
+
+            buttonView = popupWindow.getContentView().findViewById(buttonIds[i]);
+            buttonView.setText(riddle.getPossibleAnswers()[i]);
+
+            if(riddle.getPossibleAnswers()[i].equals(riddle.getAnswer()))
+                correctId = buttonIds[i];
+        }*/
+
+        popupWindow.getContentView().findViewById(correctId).setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                popupWindow.dismiss();
+            }
+        });
+
+    }
+
+    /**
+     *
+     * @param view Current view before click
+     * @param focusable Whether it can be dismissed by clicking outside the popup window
+     * @param layoutId Id of the popup layout that will be used
+     */
+    public PopupWindow onButtonShowPopupWindowClick(View view, Boolean focusable, int layoutId) {
```

Comment:
Same here, I don't think it should be here

Reply:
removed